pkg/webhook/webhook_test.go

```diff
@@ -225,3 +235,200 @@ func TestGitHubPingWebhook(t *testing.T) {
 		t.Errorf("handler returned unexpected body: got %v want %v", rr.Body, expectedResponse)
 	}
 }
+
+func TestAuthErrorCodes(t *testing.T) {
+	tests := map[string]struct {
+		err               error
+		expectedErrorCode int
+	}{
+		"gogs-verification": {
+			err:               gogs.ErrHMACVerificationFailed,
+			expectedErrorCode: 401,
+		},
+		"gogs-no-verification": {
+			err:               gogs.ErrInvalidHTTPMethod,
```

Comment:
nit: those cases about invalid HTTP methods (same comment about L289) could expect 405, or `http.StatusMethodNotAllowed`, if we wanted to be picky about this ;)

Reply:
I didn't do it because those are 3rd party provider errors and there are quite a few and they are repeated in all packages, and I didn't want to add more complexity to this.
Also the user can't do much about those errors (I would expect Github or Gitlab to send valid payloads) 
The authentication or bad secret or token one is, maybe the only error that the user can do something about it to fix it. 

I've added the 405 case and the rest are still 500.

Reply:
Agreed, git hosting providers should issue requests with valid methods and payloads. Thanks for this update, at least in the unlikely event that they don't, users will have access to more details about what's wrong.